witms/pipelines.py

```diff
@@ -17,20 +17,28 @@ def from_crawler(cls, crawler):
         settings = crawler.settings
         index_name = settings.get("ES_INDEX_NAME")
         es_hosts = settings.get("ES_HOSTS")
-        return cls(index_name, es_hosts)
+        es_buffer_size = settings.get("ES_BUFFER_SIZE")
+        return cls(index_name, es_hosts, es_buffer_size)
 
-    def __init__(self, es_index, es_hosts="localhost"):
+    def __init__(self, es_index, es_hosts="localhost", buffer_size=500):
         self.items_buffer = []
         self.es_index = es_index
+        self.buffer_size = buffer_size
         self.es_hosts = [host.strip() for host in es_hosts.split(",")]
 
     def open_spider(self, spider):
         self.client = Elasticsearch(hosts=self.es_hosts)
 
+    def close_spider(self, spider):
+        if len(self.items_buffer) > 0:
+            helpers.bulk(self.client, self.items_buffer)
+
     def process_item(self, item, spider):
+        if not item.get("publish_timestamp"):
```

Comment:
What's the purpose of this check? Are you considering that BBC articles usually do not have a publish_timestamp field, as per [this comment](https://github.com/mwesthelle/what_is_the_media_saying/blob/17f1828a8b731928d461b445ba893f7222afd4d5/witms/spiders/bbc.py#L28)? Please, document your intentions here using a comment in the code.

Reply:
I'm considering there are pages such as [this](https://www.theguardian.com/music/grammy-awards-2021) that aren't news, so we need ways to distinguish them; at first I thought about using the _authors_ field, but turns out this isn't reliable as well (see [this one](https://www.theguardian.com/world/2021/mar/15/alexei-navalny-moved-to-concentration-camp-known-for-strict-control)). Considering BBC usually doesn't have a _publish_timestamp_ field, what do you reckon we should do?

Reply:
Checked offline with @D-to-the-K . Decided to check either of `publish_timestamp`, `update_timestamp`, or `authors`, as well as document it in the code.